### PR TITLE
Fixes nxos_vpc_interface

### LINKED ISSUE #25514: nxos_vpc_interface errors out for transport nxapi
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
 - nxos_vpc_interface

##### NXOS VERSION
7.0(3)I3(1)

##### ANSIBLE VERSION
```
ansible-playbook 2.4.0 (devel 781bb44b02) last updated 2017/06/08 15:20:16 (GMT -400)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/usr/share/ansible']
  ansible python module location = /root/agents-ci/ansible/lib/ansible
  executable location = /root/agents-ci/ansible/bin/ansible-playbook
  python version = 2.7.6 (default, Oct 26 2016, 20:30:19) [GCC 4.8.4]
```

##### OS / ENVIRONMENT
running on Ubuntu 14.04

##### SUMMARY
The module errors out in latest `devel` when using transport `nxapi`.

##### STEPS TO REPRODUCE
```yaml
---
- debug: msg="START TRANSPORT:NXAPI nxos_vpc_interface sanity test"

- name: Configure vpc port channel
  nxos_vpc_interface:
    portchannel: 10
    vpc: 100
    provider: "{{ nxapi }}"

- debug: msg="END TRANSPORT:NXAPI nxos_vpc_interface sanity test"

```

##### EXPECTED RESULTS
Expect module to not error out.

##### ACTUAL RESULTS FOR TRANSPORT NXAPI

```shell
TASK [nxos_vpc_interface : debug] *************************************************************************************************************************************************************************************************************
task path: /root/agents-ci/ansible/test/integration/targets/nxos_vpc_interface/tests/nxapi/sanity.yaml:2
ok: [n9k] => {
    "failed": false, 
    "msg": "START TRANSPORT:NXAPI nxos_vpc_interface sanity test"
}

TASK [nxos_vpc_interface : Configure vpc port channel] ****************************************************************************************************************************************************************************************
task path: /root/agents-ci/ansible/test/integration/targets/nxos_vpc_interface/tests/nxapi/sanity.yaml:4
<n9k> connection transport is nxapi
Using module_utils file /root/agents-ci/ansible/lib/ansible/module_utils/basic.py
Using module_utils file /root/agents-ci/ansible/lib/ansible/module_utils/nxos.py
Using module_utils file /root/agents-ci/ansible/lib/ansible/module_utils/_text.py
Using module_utils file /root/agents-ci/ansible/lib/ansible/module_utils/pycompat24.py
Using module_utils file /root/agents-ci/ansible/lib/ansible/module_utils/six/__init__.py
Using module_utils file /root/agents-ci/ansible/lib/ansible/module_utils/six/_six.py
Using module_utils file /root/agents-ci/ansible/lib/ansible/module_utils/connection.py
Using module_utils file /root/agents-ci/ansible/lib/ansible/module_utils/network_common.py
Using module_utils file /root/agents-ci/ansible/lib/ansible/module_utils/urls.py
Using module file /root/agents-ci/ansible/lib/ansible/modules/network/nxos/nxos_vpc_interface.py
<n9k> ESTABLISH LOCAL CONNECTION FOR USER: root
<n9k> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1496954312.64-212198553137167 `" && echo ansible-tmp-1496954312.64-212198553137167="` echo $HOME/.ansible/tmp/ansible-tmp-1496954312.64-212198553137167 `" ) && sleep 0'
<n9k> PUT /tmp/tmpDoGV5P TO /root/.ansible/tmp/ansible-tmp-1496954312.64-212198553137167/nxos_vpc_interface.py
<n9k> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1496954312.64-212198553137167/ /root/.ansible/tmp/ansible-tmp-1496954312.64-212198553137167/nxos_vpc_interface.py && sleep 0'
<n9k> EXEC /bin/sh -c '/usr/bin/python /root/.ansible/tmp/ansible-tmp-1496954312.64-212198553137167/nxos_vpc_interface.py; rm -rf "/root/.ansible/tmp/ansible-tmp-1496954312.64-212198553137167/" > /dev/null 2>&1 && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_MKyy_n/ansible_module_nxos_vpc_interface.py", line 381, in <module>
    main()
  File "/tmp/ansible_MKyy_n/ansible_module_nxos_vpc_interface.py", line 288, in main
    if portchannel not in get_portchannel_list(module):
  File "/tmp/ansible_MKyy_n/ansible_module_nxos_vpc_interface.py", line 136, in get_portchannel_list
    pc_list = body[0]['TABLE_channel']['ROW_channel']
TypeError: string indices must be integers

fatal: [n9k]: FAILED! => {
    "changed": false, 
    "failed": true, 
    "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_MKyy_n/ansible_module_nxos_vpc_interface.py\", line 381, in <module>\n    main()\n  File \"/tmp/ansible_MKyy_n/ansible_module_nxos_vpc_interface.py\", line 288, in main\n    if portchannel not in get_portchannel_list(module):\n  File \"/tmp/ansible_MKyy_n/ansible_module_nxos_vpc_interface.py\", line 136, in get_portchannel_list\n    pc_list = body[0]['TABLE_channel']['ROW_channel']\nTypeError: string indices must be integers\n", 
    "module_stdout": "", 
    "msg": "MODULE FAILURE", 
    "rc": 0
}
	to retry, use: --limit @/root/agents-ci/ansible/test/integration/nxos.retry

PLAY RECAP ************************************************************************************************************************************************************************************************************************************
n9k          : ok=6    changed=0    unreachable=0    failed=1   
```

### PR DESCRIPTION
**Pull request by trishnaguha**
##### SUMMARY

- Refactor nxos_vpc_interface
- Fixes #25514 

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

modules/network/nxos/nxos_vpc_interface
test/units/modules/network/nxos/test_nxos_vpc_interface
##### ANSIBLE VERSION

```
devel 2.4
```

### CODE REVIEW

File: lib/ansible/modules/network/nxos/nxos_vpc_interface.py [hunk 1]
```diff
@@ -319,64 +295,34 @@ def main():
...
+            state_present(portchannel, delta, config_value, existing)
+
+    elif state == 'absent' and existing:
+        commands = state_absent(portchannel, existing)
 
     cmds = flatten_list(commands)
     if cmds:
         if module.check_mode:
-            module.exit_json(changed=True, commands=cmds)
+            module.exit_json(**results)
```
Comment (Qalthos) [inline lib/ansible/modules/network/nxos/nxos_vpc_interface.py]: This doesn't have commands in results yet

File: lib/ansible/modules/network/nxos/nxos_vpc_interface.py [hunk 2]
```diff
@@ -319,64 +295,34 @@ def main():
...
-    elif state == 'absent':
-        if existing.get('vpc'):
-            command = ['no vpc']
-            commands.append(command)
-        elif existing.get('peer-link'):
-            command = ['no vpc peer-link']
-            commands.append(command)
-        if commands:
-            commands.insert(0, ['interface port-channel{0}'.format(portchannel)])
+            state_present(portchannel, delta, config_value, existing)
```
Comment (Qalthos) [inline lib/ansible/modules/network/nxos/nxos_vpc_interface.py]: This returns commands but doesn't save them anywhere?

**Review by Qalthos — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/nxos/nxos_vpc_interface.py (hunk 2 above)
Comment (trishnaguha) [reply lib/ansible/modules/network/nxos/nxos_vpc_interface.py]: Right. Maybe I missed that.

**Comment by trishnaguha** [comment]
@Qalthos updated now!

**Review by Qalthos — APPROVED** [review]

**Pull request merged**